### ISSUE TITLE
Running Texamator on Mac

### ISSUE THREAD

**Issue by AmanSaurav** [alexisflesch/texamator#3]
TLDR: Would it be possible for you to create a docker app for it or atleast tell us the step by step for mac installation? coz i cant seem to make it work on it. 

Hi, i am unable to run texamator on mac.

here are the issues that i am facing. 
i tried running

`/usr/bin/python3 texamator.py`

this gave me the following error

```
Traceback (most recent call last):
  File "texamator.py", line 5, in <module>
    from PyQt5 import QtCore, QtGui, QtWidgets
ModuleNotFoundError: No module named 'PyQt5'
```

i tried installing pyqt using brew install pyqt but no success

i then tried 
`/usr/bin/python3 -m pip install python-poppler-qt5`

it kept showing an error of
` ModuleNotFoundError: No module named 'sipdistutils'`

i tried to follow [this](https://github.com/frescobaldi/python-poppler-qt5/issues/14)

 but again no success.

I know this is more of an environment issue rather than a texamator issue, but at this point i have invested close to 6 hours trying to sort it out. 

Would it be possible for you to create a docker app for it or atleast tell us the step by step for mac installation? coz i cant seem to make it work on it.

**Comment by alexisflesch** [comment]
Hi,

I understand your frustration but I don't have a Mac at home so there really is nothing I can do for you. Plus I have no experience in making docker apps.

Have you tried this ? https://stackoverflow.com/questions/59551956/how-install-python3-poppler-qt5-on-my-mac

I've had a feedback from a Mac user a while ago that told me he was able to install python-poppler-qt5 using Macports : did you try going down this path ?

I nothing works please send me an email : I'll get you in touch with that mac user and hopefully you will be able to figure it out and post a solution here.

Alexis.

**Comment by AmanSaurav** [comment]
hi, thanks for the lightning fast reply! i have tried the second link, but not the first. i will try again, tomorrow.  would it be possible for you to make a zip file of your virtualenv in which it works and send it over? probably that might work?

**Comment by alexisflesch** [comment]
Sorry for the second link, I read your question too fast. I'm not using any virtualenv on my computer : I'm not even sure that would transfer from a linux (Ubuntu 20.04) to a Mac but I can give it a try if you want.

**Comment by AmanSaurav** [comment]
even i am not sure if virtualenv would work, but its worth a try.

**Comment by alexisflesch** [comment]
venv seems pointless as pip will be asked to download the binaries on your mac : https://stackoverflow.com/questions/7450835/virtual-environment-from-mac-to-linux

You'll have to try the "Macports" solution ! Keep me posted

**Comment by AmanSaurav** [comment]
hey,
i have tried tons, but i still cant get it to work on my mac. could you convert it into a standalone app, or atleast create a docker image. that would be awesome. i keep facing issues with the poppler part.
perhaps if i could use the previous version (without poppler), that might also work.

**Comment by alexisflesch** [comment]
The old version relied on PyKDE4 which is probably harder to get working on Mac OS. Creating a docker won't help as far as my understanding goes. I don't have a mac so I'm afraid I can't do anything for you at the moment. You should seek help here : https://pypi.org/project/python-poppler-qt5/
as your problem comes from this package and not texamator. I'll let this issue open so you can post an answer if you find one.

**Comment by AmanSaurav** [comment]
Hey alexis,
I managed to get texamator running by using virtualenv of conda.. 
I was repeatedly getting issues of "sip" while using virtualenv of python.

So just in case someone else comes looking , this is what i did to get texamator running on my MacOS

1. install miniconda from [here](https://docs.conda.io/en/latest/miniconda.html)
2. create a virtualenv and activate it  using conda  `conda create --name myenv01 python=3.9.7` and `conda activate myenv01`
3. install required packages `conda install -c anaconda pyqt` and `pip install python-poppler-qt5`
4. run texamator using `python3 texamator.py`

I have written the steps from memory, so i might be a step or two, off but the important thing being , using conda running this program is way way easier. 
@alexisflesch  you may close this issue.
